interceptors/retry/retry.go

```diff
@@ -48,6 +48,7 @@ func UnaryClientInterceptor(optFuncs ...CallOption) grpc.UnaryClientInterceptor
 			if lastErr == nil {
 				return nil
 			}
+			callOpts.onRetryCallback(callCtx, attempt, lastErr)
 			logTrace(parentCtx, "grpc_retry attempt: %d, got err: %v", attempt, lastErr)
```

Comment:
Maybe we can make this log here the default retry callback behaviour while we're here?

Reply:
I can do it, but If we do this, we will lose this trace log once the custom callback will be set bu user.

Reply:
We'll document that as the default behaviour so that the user knows that it will disappear if they override the callback. They can always manually add it back again.

Reply:
I can also put this in the `WithOnRetryCallback` before invoking the user's logic, if this makes sense.

Reply:
No I think one of the things I want to do here is allow the user to disable this log behaviour too.

Reply:
All right. Done